src/compiler/utilities.ts

```diff
@@ -4751,11 +4750,6 @@ export function getTypeParameterFromJsDoc(node: TypeParameterDeclaration & { par
     return typeParameters && find(typeParameters, p => p.name.escapedText === name);
 }
 
-/** @internal @knipignore */
-export function hasTypeArguments(node: Node): node is HasTypeArguments {
-    return !!(node as HasTypeArguments).typeArguments;
-}

```

Comment:
This internal utility function was unused, and I noticed it has buggy guarding semantics which tripped me up when I tried to use it:

- `node` could be a `NodeWithTypeArguments`, which is not a member of `HasTypeArguments`
- else branch narrowing was wrong: `typeArguments` is optional, so it can be undefined while `node` is still assignable to `HasTypeArguments`

The former is trivially fixable while the latter is not, so I opted to purge this to avoid future confusion. Let me know if I should do something else instead.